core/Pos/Binary/Class/Numbers.hs

```diff
@@ -204,156 +65,69 @@ newtype TinyVarInt = TinyVarInt {getTinyVarInt :: Word16}
 newtype FixedSizeInt a = FixedSizeInt {getFixedSizeInt :: a}
     deriving (Eq, Ord, Show, Generic, NFData, Functor, Enum, Num, Integral, Real)
 
-instance TypeError
-    ('Text "Do not encode 'Int' directly. Instead, use one of newtype wrappers:" ':$$:
-     'Text "  'FixedSizeInt': always uses 8 bytes" ':$$:
-     'Text "  'SignedVarInt': uses 1–10 bytes (1 byte for −64..63)" ':$$:
-     'Text "  'UnsignedVarInt': uses 1–10 bytes (1 byte for 0..127);" ':$$:
-     'Text "                    more efficient for non-negative numbers," ':$$:
-     'Text "                    but takes 10 bytes for negative numbers")
-  => Bi Int
-  where
-    get = error "get@Int"
-    put = error "put@Int"
-    size = error "size@Int"
-
-instance TypeError
-    ('Text "Do not encode 'Word' directly. Instead, use one of newtype wrappers:" ':$$:
-     'Text "  'FixedSizeInt': always uses 8 bytes" ':$$:
-     'Text "  'UnsignedVarInt': uses 1–10 bytes (1 byte for 0..127)")
-  => Bi Word
-  where
-    get = error "get@Word"
-    put = error "put@Word"
-    size = error "size@Word"
-
 -- Int
 
 instance Bi (UnsignedVarInt Int) where
-    put (UnsignedVarInt a) = putUnsignedVarInt (fromIntegral a :: Word)
-    {-# INLINE put #-}
-    get = label "UnsignedVarInt Int" $
-        UnsignedVarInt . (fromIntegral :: Word -> Int) <$>
-        limitGet 15 getUnsignedVarInt'
-    {-# INLINE get #-}
-    size = VarSize $ \(UnsignedVarInt a) ->
-              getUnsignedVarIntSize (fromIntegral a :: Word)
-    {-# INLINE size #-}
+  encode = encode . getUnsignedVarInt
```

Comment:
For now, to not have to do drastic changes to the whole codebase, we are simply offloading encoding and decoding of all these wrappers to the encoding/decoding of the inner type they wrap.

In the future I guess we can review if these are needed anymore (see note on `tinyVarInt`).